EnhancedMediaPlayer/MediaPlayerControlsView.swift

```diff
@@ -0,0 +1,149 @@
+// MediaPlayerControlsView.swift
+
+import SwiftUI
+
+struct MediaPlayerControlsView: View {
+    @ObservedObject var viewModel: ViewModel
+
+    var body: some View {
+        renderOverlay()
+        renderActionButtons()
+            .padding(.horizontal, viewModel.screenWidth / Constants.paddingWidthDivider)
+    }
+
+    @ViewBuilder private func renderOverlay() -> some View {
+        Rectangle()
+            .fill(.black)
+            .opacity(Constants.overlayOpacity)
+            .frame(maxWidth: .infinity, maxHeight: .infinity)
+    }
+
+    @ViewBuilder private func renderActionButtons() -> some View {
+        HStack(spacing: .zero) {
+            renderRewind()
+                .frame(maxWidth: .infinity, alignment: .leading)
+            if viewModel.playerState == .finished {
+                renderReplay()
+            } else {
+                renderPlayPause()
+            }
+            renderForward()
+                .frame(maxWidth: .infinity, alignment: .trailing)
+        }
+        .frame(maxWidth: .infinity)
+    }
+
+    @ViewBuilder private func renderPlayPause() -> some View {
+        renderButton(control: viewModel.playerState == .playing ? .pause : .play)
+    }
+
+    @ViewBuilder private func renderReplay() -> some View {
+        renderButton(control: .replay)
+    }
+
+    @ViewBuilder private func renderRewind() -> some View {
+        renderTimePlaceholderWrapper {
+            renderButton(control: .rewind)
+        }
+    }
+
+    @ViewBuilder private func renderForward() -> some View {
+        renderTimePlaceholderWrapper {
+            renderButton(control: .forward)
+        }
+    }
+
+    @ViewBuilder private func renderButton(control: Control) -> some View {
+        control.systemImage
+            .resizable()
+            .frame(width: Constants.iconSize.width, height: Constants.iconSize.height)
+            .foregroundColor(.white)
+            .wrapInButton(action: { viewModel.onTapAction?(control) })
+    }
+
+    @ViewBuilder private func renderTimePlaceholderWrapper<Content: View>(
+        @ViewBuilder content: @escaping () -> Content
+    ) -> some View {
+        ZStack {
+            content()
+            Text(viewModel.formattedSeekFactor)
+                .foregroundColor(.white)
+                .font(.system(size: Constants.timeFontSize))
+        }
+    }
+}
+
+extension MediaPlayerControlsView {
+    class ViewModel: ObservableObject {
+        @Binding var screenWidth: CGFloat
+        @Binding var playerState: MediaPlayerView.PlayerState
+
+        var onTapAction: ((Control) -> Void)?
```

Comment:
This will never be changed after this `ViewModel` is instantiated, it shouldn't be a var, and it should also not be nullable. Btw in `ViewModels` that aren't for screens (they aren't the ones managed by the `Controller`, I prefer to not use actions, and put the action in the `View` itself. So to instantiate this one you'd do this:
```swift
MediaPlayerControlsView(
    viewModel: .init(screenWidth: $something, ...),
    onTapAction: { /* something*/ }
)
```
If you want you can leave it in the `ViewModel`, no problem, but definitely make it a `let onTapAction: (Control) -> Void`